src/zeit/cms/repository/browser/menu.py

```diff
@@ -29,8 +29,15 @@ class Delete(zeit.cms.browser.menu.LightboxActionMenuItem):
 
     @property
     def visible(self):
-        return IRepositoryContent.providedBy(self.context) and \
-            not IPublishInfo(self.context).published
+        if not IRepositoryContent.providedBy(self.context):
+            return False
+        elif IPublishInfo(self.context).published:
+            return False
+        elif IFolder.providedBy(self.context):
+            for item in self.context.values():
+                if IPublishInfo(item).published:
```

Comment:
ftr, we've worked around this in c9cb0d600d1516580c3f87515791dd3eeff181f9 — now only the actual 'delete' view performs that check, but not the (much more frequently used) main view.